app/hotels/src/singleHotel/summary/SummaryRow.js

```diff
@@ -15,49 +14,38 @@ import { defaultTokens } from '@kiwicom/mobile-orbit';
 type Props = {|
   +text: React.Element<typeof Translation>,
   +price: React.Element<typeof Price>,
-  +width: number,
 |};
 
-type State = {|
-  +maxTextWidth: number | null,
-|};
+const SPACING = 5;
 
-class SummaryRow extends React.Component<Props, State> {
-  state = {
-    maxTextWidth: null,
-  };
+const SummaryRow = ({ text, price }: Props) => {
+  const [containerWidth, setContainerWidth] = React.useState(null);
+  const [priceWidth, setPriceWidth] = React.useState(null);
 
-  onLayout = (event: OnLayout) => {
-    const pagePadding = 34;
-    const spacing = 5;
-    // Text does not play well with price when it grows too long.
-    // Wee need to calculate maxWidth for the text, unless it will push
-    // Price out of the screen
-    this.setState({
-      maxTextWidth:
-        this.props.width -
-        event.nativeEvent.layout.width -
-        pagePadding -
-        spacing,
-    });
-  };
+  function onContainerLayout(event: OnLayout) {
+    setContainerWidth(event.nativeEvent.layout.width);
+  }
 
-  render() {
-    return (
-      <View style={styles.row}>
-        <Text
-          style={[styles.text, { maxWidth: this.state.maxTextWidth }]}
-          numberOfLines={1}
-        >
-          {this.props.text}
-        </Text>
-        <View onLayout={this.onLayout}>
-          <Text style={styles.price}>{this.props.price}</Text>
-        </View>
-      </View>
-    );
+  function onPriceLayout(event: OnLayout) {
+    setPriceWidth(event.nativeEvent.layout.width);
   }
-}
+
+  const maxWidth =
+    containerWidth != null && priceWidth != null
+      ? containerWidth - priceWidth - SPACING
+      : 0;
```

Comment:
Does it behave correctly when the max width is zero?

Reply:
Setting the max width to zero just displays the price on the right hand-side, and the text is obviously not showing. That's an acceptable behaviour for me (especially when it should not happen after the `View`s rendered and executed their `onLayout`).

![Screenshot 2019-05-07 at 09 59 24](https://user-images.githubusercontent.com/22741774/57283318-2e4e9380-70af-11e9-8fbb-62e31d67918c.png)
![Screenshot 2019-05-07 at 10 01 00](https://user-images.githubusercontent.com/22741774/57283319-2ee72a00-70af-11e9-966f-e1aebeba9cf2.png)